src/rabbit_amqqueue.erl

```diff
@@ -397,6 +397,11 @@ not_found_or_absent_dirty(Name) ->
     end.
 
 with(Name, F, E) ->
+    with(Name, F, E, 1000).
+
+with(_Name, _F, E, 0) ->
+    E(infinite_loop_detected);
```

Comment:
Should we make `with_or_die/2` handle this case gracefully? at least with something that other `function_clause`?